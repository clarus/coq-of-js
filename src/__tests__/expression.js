// @flow
import {compileAndPrint} from "../compiler/index.js";

it("requires function parameters to be simple names", () => {
  expect(compileAndPrint(`function foo({a}) {return a;}`)).toMatchSnapshot();
});

it("handles empty arrays", () => {
  expect(compileAndPrint(`const a = [];`)).toMatchSnapshot();
});

it("does not handle empty elements in arrays", () => {
  expect(compileAndPrint(`const a = [,];`)).toMatchSnapshot();
});

it("does not handle spreads in arrays", () => {
  expect(compileAndPrint(`const a = [...[]];`)).toMatchSnapshot();
});

it("does not handle calls with spread parameters", () => {
  expect(compileAndPrint(`const n = f(...a);`)).toMatchSnapshot();
});

it("handles nulls", () => {
  expect(compileAndPrint(`const n = null;`)).toMatchSnapshot();
});

it("handles empty objects", () => {
  expect(compileAndPrint(`const o = {};`)).toMatchSnapshot();
});

it("does not handle non-objects without annotations", () => {
  expect(compileAndPrint(`const o = {x: 12};`)).toMatchSnapshot();
});

it("handles enum instances", () => {
  expect(compileAndPrint(`const e = ("Green": Kind);`)).toMatchSnapshot();
});
