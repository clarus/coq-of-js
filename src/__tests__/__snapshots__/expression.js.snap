// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not handle calls with spread parameters 1`] = `
"> 1 | const n = f(...a);
    |            ^^^^

Unhandled function argument"
`;

exports[`does not handle empty elements in arrays 1`] = `
"> 1 | const a = [,];
    |          ^^^

Expected non-empty elements in the array"
`;

exports[`does not handle non-empty objects without annotations 1`] = `
"> 1 | const o = {x: 12};
    |          ^^^^^^^

Unhandled object expression without type annotation"
`;

exports[`does not handle records with computed property names 1`] = `
"> 1 | const o = ({[a]: \\"hi\\"}: Rec);
    |             ^

Unhandled computed property name"
`;

exports[`does not handle records with methods 1`] = `
"> 1 | const o = ({foo() {}}: Rec);
    |            ^^^^^^^^

Unhandled method element in object"
`;

exports[`does not handle records with spreads 1`] = `
"> 1 | const o = ({...rec}: Rec);
    |            ^^^^^^

Unhandled spread in object"
`;

exports[`does not handle spreads in arrays 1`] = `
"> 1 | const a = [...[]];
    |           ^^^^^

Spreads in arrays are not handled"
`;

exports[`handles empty arrays 1`] = `
"Definition a :=
  []."
`;

exports[`handles empty objects 1`] = `
"Definition o :=
  tt."
`;

exports[`handles enum instances 1`] = `
"Definition e :=
  Kind.Green."
`;

exports[`handles nulls 1`] = `
"Definition n :=
  tt."
`;

exports[`handles record instances 1`] = `
"Definition o :=
  {| Rec.a := \\"hi\\"; Rec.b := 12; |}."
`;

exports[`requires function parameters to be simple names 1`] = `
"> 1 | function foo({a}) {return a;}
    |             ^^^

Expected simple identifier as function parameter"
`;
